integrations/mistral/src/haystack_integrations/components/converters/mistral/ocr_document_converter.py

```diff
@@ -0,0 +1,367 @@
+                - `meta`: Aggregated metadata dictionary with structure:
+                  {"source_page_count": int, "source_total_images": int, "source_*": any}.
+                  If document_annotation_schema was provided, all annotation fields are unpacked
+                  with 'source_' prefix (e.g., source_language, source_chapter_titles, source_urls).
+            - `raw_mistral_response`:
+                List of dictionaries containing raw OCR responses from Mistral API (one per source).
+                Each response includes per-page details, images, annotations, and usage info.
+        """
+        # Convert Pydantic models to Mistral ResponseFormat schemas
+        bbox_annotation_format = (
+            response_format_from_pydantic_model(bbox_annotation_schema) if bbox_annotation_schema else None
+        )
+        document_annotation_format = (
+            response_format_from_pydantic_model(document_annotation_schema) if document_annotation_schema else None
+        )
+
+        # Process each source
+        documents = []
+        raw_responses = []
+        uploaded_file_ids = []  # Track files we uploaded for cleanup
+
+        try:
+            for source in sources:
+                try:
+                    # Convert source to Mistral chunk format
+                    chunk = self._convert_source_to_chunk(source)
+
+                    # Track uploaded files (only those we created, not user-provided FileChunks)
+                    if isinstance(source, (str, Path, ByteStream)) and isinstance(chunk, FileChunk):
+                        uploaded_file_ids.append(chunk.file_id)
+
+                    # Call Mistral OCR API with the provided source
+                    ocr_response: OCRResponse = self.client.ocr.process(
+                        model=self.model,
+                        document=chunk,
+                        include_image_base64=self.include_image_base64,
+                        pages=self.pages,
+                        image_limit=self.image_limit,
+                        image_min_size=self.image_min_size,
+                        bbox_annotation_format=bbox_annotation_format,
+                        document_annotation_format=document_annotation_format,
+                    )
+
+                    # Process the OCR response into a Document
+                    document = self._process_ocr_response(ocr_response, document_annotation_schema)
+                    documents.append(document)
+                    raw_responses.append(ocr_response.model_dump())
+                except Exception as e:
+                    logger.warning(
+                        "Could not process source {source}. Skipping it. Error: {error}",
+                        source=source,
+                        error=e,
+                    )
+                    continue
+        finally:
+            # Cleanup uploaded files if enabled
+            if self.cleanup_uploaded_files and uploaded_file_ids:
```

Comment:
I see that you introduced this option to delete uploaded files. Could you please explain why you think it's necessary/helpful for a user?

Reply:
Ah, you already did this. Great!